FWCore/Modules/test/testGenericConsumer.py

```diff
@@ -0,0 +1,37 @@
+# Configuration file for EventContentAnalyzer
+
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("TEST")
+
+process.load("FWCore.Framework.test.cmsExceptionsFatal_cff")
+process.options.wantSummary = True
+
+process.load("FWCore.Modules.printContent_cfi")
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(5)
+)
+
+process.source = cms.Source("EmptySource")
+
+process.thing = cms.EDProducer("ThingProducer")
```

Comment:
I'd expect this test to succeed even if the consumes connections from `GenericConsumer` to the producer modules would somehow break. For the `Int` producers we already have `edmtest::MustRunIntProducer` that throws an exception if the module's Event transition was not run. An alternative would be to add the Tracer and compare logs, but such tests tend to be brittle.

Reply:
IMHO this is the fundamental problem we run into with the unscheduled execution: anything that wants to _test_ if a module has run may _cause_ it to run.

Reply:
I've updated the test with all the trickeries needed to actually test that the `GenericConsumer` is working as intended...

@makortel could you still "fix" the `ConditionalTask` even if HLT may not use it next year ?

Reply:
I'd prefer to have a simpler test that does not add dependence on the behavior of `ConditionalTask`. One additional simple alternative for inspecting whether a module was ran or not would be to use the `process.options.wantSummary = True` printout. At this point I could be willing to rework this test configuration in a subsequent PR though.

> could you still "fix" the `ConditionalTask` even if HLT may not use it next year ?

I'm afraid addressing the behavior of modules in `ConditionalTask` that are not consumed the associated `Path`(s) is very low in the priority list without a real use case.

Reply:
> One additional simple alternative for inspecting whether a module was ran or not would be to use the `process.options.wantSummary = True` printout.

Do you mean inspecting by eye ?
Or piping the out through `grep`, _etc_ ?
That seems to me more fragile that depending on the framework behaviour.

Reply:
Done with a shell script that runs the test, though I'm not much a fan of this approach.

Reply:
> Or piping the out through `grep`, _etc_ ?

Some approach where a script checks the correctness, yes.

I agree that the job itself failing in case of misbehavior would be better. That behavior could be achieved e.g. with the `edmtest::MustRunIntProducer` that I mentioned earlier (that would have been the direction I would have evolved this test). In principle we could think of crafting a specific Service to enforce that a given module is run for a given transition for a given number of times (a thought from @Dr15Jones).